pennylane_lightning/src/benchmarks/Bench_ApplyMyImpl.cpp

```diff
@@ -0,0 +1,169 @@
+ */
+template <class T>
+static void applyOperation_MyImplPauliX(benchmark::State &state, Kernel kernel) {
+    const size_t num_qubits = state.range(0);
+    const size_t num_wires = state.range(1);
+
+    if (!num_qubits) {
+        state.SkipWithError("Invalid number of qubits.");
+    }
+    if (!num_wires) {
+        state.SkipWithError("Invalid number of wires.");
+    }
+
+    for (auto _ : state) {
+        Pennylane::StateVectorManaged<T> sv{num_qubits};
+
+        sv.applyOperation(kernel, "PauliX", {num_wires}, false);
+
+        benchmark::DoNotOptimize(sv.getDataVector()[0]);
+        benchmark::DoNotOptimize(sv.getDataVector()[(1 << num_qubits) - 1]);
+    }
+}
+
+
+template <class T>
+static void applyOperation_MyImplPauliZ(benchmark::State &state, Kernel kernel) {
+    const size_t num_qubits = state.range(0);
+    const size_t num_wires = state.range(1);
+
+    if (!num_qubits) {
+        state.SkipWithError("Invalid number of qubits.");
+    }
+    if (!num_wires) {
+        state.SkipWithError("Invalid number of wires.");
+    }
+
+    for (auto _ : state) {
+        Pennylane::StateVectorManaged<T> sv{num_qubits};
+
+        sv.applyOperation(kernel, "PauliZ", {num_wires}, false);
+
+        benchmark::DoNotOptimize(sv.getDataVector()[0]);
+        benchmark::DoNotOptimize(sv.getDataVector()[(1 << num_qubits) - 1]);
+    }
+}
+
+template <class T>
+static void applyOperation_PauliXLM(benchmark::State &state, Kernel kernel) {
+    const size_t num_qubits = state.range(0);
+    const size_t num_wires = state.range(1);
+
+    if (!num_qubits) {
+        state.SkipWithError("Invalid number of qubits.");
+    }
+    if (!num_wires) {
+        state.SkipWithError("Invalid number of wires.");
+    }
+
+    for (auto _ : state) {
+        Pennylane::StateVectorManaged<T> sv{num_qubits};
+
+        sv.applyOperation(kernel, "PauliX", {num_wires}, false);
+
+        benchmark::DoNotOptimize(sv.getDataVector()[0]);
+        benchmark::DoNotOptimize(sv.getDataVector()[(1 << num_qubits) - 1]);
+    }
+}
+
+template <class T>
+static void applyOperation_PauliZLM(benchmark::State &state, Kernel kernel) {
+    const size_t num_qubits = state.range(0);
+    const size_t num_wires = state.range(1);
+
+    if (!num_qubits) {
+        state.SkipWithError("Invalid number of qubits.");
+    }
+    if (!num_wires) {
+        state.SkipWithError("Invalid number of wires.");
+    }
+
+    for (auto _ : state) {
+        Pennylane::StateVectorManaged<T> sv{num_qubits};
+
+        sv.applyOperation(kernel, "PauliZ", {num_wires}, false);
+
+        benchmark::DoNotOptimize(sv.getDataVector()[0]);
+        benchmark::DoNotOptimize(sv.getDataVector()[(1 << num_qubits) - 1]);
+    }
+}
+
+BENCHMARK_APPLYOPS(applyOperation_PauliXLM, float, LM, Kernel::LM)
+    ->ArgsProduct({
+        benchmark::CreateDenseRange(6, 30, /*step=*/6), // num_qubits
+        benchmark::CreateDenseRange(2, 4, /*step=*/1),  // num_wires
```

Comment:
@multiphaseCFD, in the benchmarks you tested the application of the gate to different wires.
What did you expect to observe with this?

Reply:
Also, it would perhaps be interesting to observe the scaling with the number of threads.

Reply:
What I expected is we can get better acceleration when then wire number is less than 4.  Increasing the number of threads would not lead to speed.